apisix/plugins/proxy-rewrite.lua

```diff
@@ -317,6 +318,8 @@ function _M.rewrite(conf, ctx)
             upstream_uri = core.utils.uri_safe_encode(upstream_uri)
         end
 
+        req_set_uri(upstream_uri)
```

Comment:
Note that we don't need to set uri when `use_real_request_uri_unsafe` is true, because in that case, the `upstream_uri` is supposed to take precedence.